src/clone_area_cmd.cpp

```diff
@@ -0,0 +1,533 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file clone_area_cmd.cpp Commands related to clone area. */
+
+#include "stdafx.h"
+#include "command_func.h"
+#include "viewport_func.h"
+#include "company_base.h"
+#include "clone_area_cmd.h"
+#include "newgrf_station.h"
+#include "safeguards.h"
+#include "road.h"
+#include "road_internal.h"
+#include "depot_base.h"
+#include "tunnelbridge_map.h"
+#include "sound_func.h"
+#include "rail_cmd.h"
+#include "tunnelbridge_cmd.h"
+#include "station_cmd.h"
+#include "terraform_cmd.h"
+
+TileIndex selected_tile;
+TileIndex selected_start_tile;
+bool selected_diagonal;
```

Comment:
Globals should be avoided if possible, and if used, need an underscore prefix, like `TileIndex _selected_tile;`.

I wonder, can these be stored as properties of `TerraformToolbarWindow`, like `last_user_action`? They'd then be passed as command parameters.

The copied area would be lost when the terraform toolbar is closed, although that is already the case (possibly just because the button is disabled when the toolbar is initialized).